Add jid 0.6.2 (New Formula)

ikeisuke opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
jid
https://github.com/simeji/jid

neutric commented on Formula/jid.rb:
```diff
@@ -0,0 +1,22 @@
+class Jid < Formula
+  desc "jid"
```
Thanks for your contribution! A more comprehensive description of what jid does would be welcome.

ikeisuke replied:
Hi, @neutric. Thank you for review.
I've updated description.

MikeMcQuaid replied:
Not seeing an updated description.

neutric commented on Formula/jid.rb:
```diff
@@ -0,0 +1,22 @@
+  end
+
+  test do
+    assert_match "jid version v#{version}", shell_output("#{bin}/jid --version")
```
A more complex test than a simple version string comparison is preferable.

neutric replied:
A more complex test than a simple version string comparison is preferable.

MikeMcQuaid replied:
No, we'd rather use `assert_match` and `(testpath/"test.json").write` with some sample JSON for a better test.

MikeMcQuaid commented on Formula/jid.rb:
```diff
@@ -0,0 +1,22 @@
+  homepage "https://github.com/simeji/jid"
+  version "0.6.2"
+  if Hardware::CPU.is_64_bit?
+    url "#{homepage}/releases/download/#{version}/jid_darwin_amd64.zip"
```
This needs to be build from source instead.

ikeisuke replied:
Hi @MikeMcQuaid . Thank you for review.
I've updated PR.

MikeMcQuaid commented on Formula/jid.rb:
```diff
@@ -0,0 +1,23 @@
+class Jid < Formula
+  desc "jid"
+  homepage "https://github.com/simeji/jid"
+  version "0.6.2"
```
Remove this.

MikeMcQuaid commented on Formula/jid.rb:
```diff
@@ -0,0 +1,23 @@
+  desc "jid"
+  homepage "https://github.com/simeji/jid"
+  version "0.6.2"
+  url "#{homepage}/archive/#{version}.zip"
```
Hardcode this, don't use variables.

MikeMcQuaid commented on Formula/jid.rb:
```diff
@@ -0,0 +1,23 @@
+  end
+
+  test do
+    assert_match "jid version v#{version}", shell_output("#{bin}/jid --version")
```
Still need a better test.

MikeMcQuaid replied:
Actually, this is fine 👍

ikeisuke commented:
@MikeMcQuaid thank you for review.

I've updated desc, removed version and change to hardcode url.

I'm considering a method of test code implementation.
I think that it is difficult to write a complex test unless update jid.

MikeMcQuaid commented on Formula/jid.rb:
```diff
@@ -0,0 +1,22 @@
+class Jid < Formula
+  desc "Json incremental digger"
+  homepage "https://github.com/simeji/jid"
+  url "https://github.com/simeji/jid/archive/0.6.2.zip"
```
`https://github.com/simeji/jid/archive/0.6.2.tar.gz`

MikeMcQuaid commented:
Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format is `jid 0.6.2 (new formula)`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md). Thanks!

neutric commented:
Thanks guys!

ikeisuke commented:
@neutric, @MikeMcQuaid 
Thanks again for your help and your review here.
